exchanges/kucoin/kucoin_websocket.go

```diff
@@ -75,8 +75,8 @@ const (
 )
 
 var subscriptionNames = map[string]string{
-	subscription.TickerChannel:    marketTickerChannel,
-	subscription.OrderbookChannel: marketOrderbookLevel2Channels,
+	subscription.TickerChannel:    marketAllTickersChannel,         // This allows more subscriptions on the orderbook channel for this specific connection.
```

Comment:
A comment that I want to investigate further. If spot or margin is not enabled, this channel shouldn't be subscribed to, its wasteful given that it doesn't contain futures data.

Reply:
In the function `expandSubscription` I think you can add the following check to it
```go
if len(assetPairs[s.Asset]) == 0 {
		return nil, nil
	}
```
That will prevent `ticker:all` being subscribed to if there are no spot pairs. I don't yet know whether ticker:all applies to margin pairs too though and may need additional changes